kornia/contrib/sam/model.py

```diff
@@ -0,0 +1,167 @@
+"""Based from the original code from Meta Platforms, Inc. and affiliates.
+
+https://github.com/facebookresearch/segment-
+anything/blob/3518c86b78b3bc9cf4fbe3d18e682fad1c79dc51/segment_anything/build_sam.py
+"""
+from __future__ import annotations
+
+from enum import Enum
+
+import torch
+
+from kornia.contrib.sam.architecture import Sam as SAM_ARCH
+from kornia.contrib.sam.architecture.common import LayerNorm
+from kornia.contrib.sam.architecture.image_encoder import ImageEncoderViT
+from kornia.contrib.sam.architecture.mask_decoder import MaskDecoder
+from kornia.contrib.sam.architecture.prompt_encoder import PromptEncoder
+from kornia.contrib.sam.architecture.transformer import TwoWayTransformer
+from kornia.contrib.sam.base import ModelBase
+
+
+class SamModelType(Enum):
+    """Map the SAM model types."""
+
+    vit_h = 0
+    vit_l = 1
+    vit_b = 2
+
+
+class Sam(ModelBase[SamModelType], SAM_ARCH):
```

Comment:
Don't use double inheritance, pass the model in the constructor or build via class method